zip/zlib.nim

```
@@ -305,8 +305,6 @@ proc uncompress*(sourceBuf: cstring, sourceLen: Natural; stream=DETECT_STREAM):
   of DETECT_STREAM: MAX_WBITS + 32
 
   var status = inflateInit2(z, wbits.int32)
-  if status != Z_OK:
-    return
```

So raise an exception instead of ``return``.

---

on [line 309](https://github.com/emekoi/zip/blob/master/zip/zlib.nim#L309) there is a case statement that decides which exception to raise is case of `status` not being `Z_OK`.